File: GeneratedEntityReaderClasses/Components/ItemCostComponent.cs
namespace GeneratedNoitaClasses;
public class ItemCostComponent : NoitaComponentBase,  iNoitaType<ItemCostComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt64 cost { get; set; } = new();
	public NoitaBool stealable { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		cost.Read(s);
		stealable.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		cost.Write(s);
		stealable.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/ItemCostComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt64

from NoitaStream import NoitaStream
@dataclass
class ItemCostComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    cost:NoitaInt64 = field(default_factory=NoitaInt64)
    stealable:NoitaBool = field(default_factory=NoitaBool)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.cost.Read(s)
        self.stealable.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.cost.Write(s)
        self.stealable.Write(s)

